### PR TITLE
feat(Reddit - Hide banner ads): Constrain to last working version

### PR DESCRIPTION
**Pull request by ascopes**
Until GH-3099 is fixable and addressed, this limits the version of Reddit that can be
used to 2024.0.17 which is the last known working version.

### CODE REVIEW

**Comment by ascopes** [comment]
Not part of this MR but possibly worth noting that the link to the docs explaining how to structure the commit naming seems to be broken, so the name of this commit is a bit of a stab in the dark on my behalf.

Hopefully this is correct.

File: src/main/kotlin/app/revanced/patches/reddit/ad/banner/HideBannerPatch.kt [hunk 1]
```diff
@@ -4,7 +4,13 @@ import app.revanced.patcher.data.ResourceContext
 import app.revanced.patcher.patch.ResourcePatch
 import app.revanced.patcher.patch.annotation.Patch
 
-@Patch(description = "Hides banner ads from comments on subreddits.")
+@Patch(
+    description = "Hides banner ads from comments on subreddits.",
+    // Pinned to 2024.0.17 until https://github.com/iBotPeaches/Apktool/issues/3534
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/reddit/ad/banner/HideBannerPatch.kt:9]:
```suggestion
    // Constrained to last working version (2024.0.17) until https://github.com/iBotPeaches/Apktool/issues/3534
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Since this patch is a dependency and not usable in any frontend, it inherits the compatibility of the dependant patches. For that reason, the compatibility needs to be set on the patches which depend on this one. Alternatively, this patch can get a name & other patches can stop depend on it so that they can be used on later versions with this patch unselected.

**Comment by ascopes** [comment]
If I go for the latter, is it just a case of setting the name attribute directly or do other patches need modifying as well?

Apologise if it is a silly question, somewhat new to how this works!

File: src/main/kotlin/app/revanced/patches/reddit/ad/banner/HideBannerPatch.kt [hunk 2]
```diff
@@ -4,7 +4,13 @@ import app.revanced.patcher.data.ResourceContext
 import app.revanced.patcher.patch.ResourcePatch
 import app.revanced.patcher.patch.annotation.Patch
 
-@Patch(description = "Hides banner ads from comments on subreddits.")
+@Patch(
+    description = "Hides banner ads from comments on subreddits.",
+    // Constrained to last working version (2024.0.17) until https://github.com/iBotPeaches/Apktool/issues/3534
+    // is addressed which is causing crashes during patching.
+    // See https://github.com/ReVanced/revanced-patches/issues/3099.
+    compatiblePackages = [CompatiblePackage("com.reddit.frontpage", ["2024.0.17"])]
```
Comment (programminghoch10) [inline src/main/kotlin/app/revanced/patches/reddit/ad/banner/HideBannerPatch.kt:12]:
```suggestion
    // Constrained to last working version (2024.17.0) until https://github.com/iBotPeaches/Apktool/issues/3534
    // is addressed which is causing crashes during patching.
    // See https://github.com/ReVanced/revanced-patches/issues/3099.
    compatiblePackages = [CompatiblePackage("com.reddit.frontpage", ["2024.17.0"])]
```
Wasn't the last working version `2024.17.0` and not `2024.0.17`?

**Review by programminghoch10 — CHANGES_REQUESTED** [review]

**Comment by ericswpark** [comment]
@ascopes why was the head repository deleted? Did another PR supersede this one?

**Comment by ascopes** [comment]
Ah whoops, I was removing an unused clone for another revanced project and accidentally deleted the wrong one... will reraise with the correct fixes as given above when I get some time!

**Comment by ascopes** [comment]
Reopened at #3192

**Pull request closed without merging**